api/src/Page/Sample.php

```diff
@@ -2022,13 +2044,13 @@ function _add_sample_to_group() {
 
             if ($chk[0]['SAMPLES'] != $in_proposal) $this->_error('No such sample group');
 
-            $order = $this->has_arg('GROUPORDER') ? $this->arg('GROUPORDER') : 1;
-            $type = $this->has_arg('TYPE') ? $this->arg('TYPE') : null;
+            $order = isset($sample_data['GROUPORDER']) ? $sample_data['GROUPORDER'] : 1;
+            $type = isset($sample_data['TYPE']) ? $sample_data['TYPE'] : null;
 
             $this->db->pq("INSERT INTO blsamplegroup_has_blsample (blsampleid, blsamplegroupid, grouporder, type) 
                 VALUES (:1,:2, :3, :4)", array($this->arg('BLSAMPLEID'), $sgid, $order, $type));
 
-            $this->_output(array('BLSAMPLEGROUPSAMPLEID' => $sgid.'-'.$this->arg('BLSAMPLEID'), 'BLSAMPLEID' => $this->arg('BLSAMPLEID'), 'BLSAMPLEGROUPID' => $sgid));
+            return array('BLSAMPLEID' => $this->arg('BLSAMPLEID'), 'BLSAMPLEGROUPID' => $sgid);
```

Comment:
Need to check if this still works with the current branch. We are not returning the hybrid samplegroupsampleid?

Also still using this->arg() here not sample_data. Not convinced the use of sample_data saves much here.  You still need to check presence of keys in the array which is what this->has_arg() does. 

Using the args/has_arg is consistent with the rest of the code.

An alternative is to setup the parameters in the calling function and pass them in as variables. So create a $blsampleId, $blsampleGroupId, $groupOrder, $type as arguments to _save_sample_to_group(). Then you define a clear method signature.

Checkout the _update_autocollect_session method in Proposal.php for an example